Code/src/main/java/nl/utwente/group10/haskell/type/Type.java

```diff
@@ -46,4 +70,93 @@ public final Type getFresh() {
     
     @Override
     public abstract String toString();
+    
+    /**
+     * The list type constructor.
+     */
+    public final static TypeCon ListCon = new ListTypeCon();
+    
+    /**
+     * @return a new type variable constructed from a name and optionally a number of class constraint 
+     * @param name Identifier for this type variable.
+     * @param constraints The constraints for this type
+     */
+    public final static TypeVar var(final String name, TypeClass... constraints) {
+        return new TypeVar(name, constraints);
+    }
+    
+    /**
+     * @return a new type variable conctructed from a name and a class constraint set. 
+     * @param name Identifier for this type variable.
+     * @param constraints The set of constraints for this type
+     */
+    public final static TypeVar var(final String name, final Set<TypeClass> constraints) {
+        return new TypeVar(name, 0, constraints, null);
+    }
+
+    /**
+     * @return a new type constructor
+     * @param The name of type constructor.
+     */
+    public final static TypeCon con(String name) {
+        return new TypeCon(name);
+    }
+
+    /**
+     * @return a new type from a applied type constructor
+     * @param The name of type constructor.
+     * @param List of type argument to constructo is applied too
+     */
+    public final static Type con(String name, Type... args) {
+        Type t = new TypeCon(name);
+        for (Type a : args) {
+            t = new TypeApp (t, a);
+        }
+        return t;
+    }
+    
+    /**
+     * @return a new function type in a chain with all arguments
+     * @param List of types 
+     */
+    public final static Type fun(Type... elems) {
+        final int last = elems.length-1;
+        Type t = elems[last];
```

Comment:
This fails with an ArrayIndexOutOfBoundsException on a zero-element elems array.